test/powershell/Modules/Microsoft.PowerShell.Utility/NewTemporaryFile.Tests.ps1

```diff
@@ -21,11 +21,25 @@ Describe "NewTemporaryFile" -Tags "CI" {
 
         Test-Path $tempFile | Should be $true
         $tempFile | Should BeOfType System.IO.FileInfo
+        $tempFile.Extension | Should be '.tmp'
 
-        if(Test-Path $tempFile)
-        {
-            Remove-Item $tempFile -ErrorAction SilentlyContinue -Force
-        }
+        Remove-Item $tempFile -ErrorAction SilentlyContinue -Force
```

Comment:
Suggestion: To make cleanup reliable, use 'try' around the Should statements and place 'Remove-Item' in the finally block.